README.Rmd

```diff
@@ -0,0 +1,114 @@
+---
+title: "Homework 5"
+author: "Ben Wang"
+date: "8/3/20"
+output: github_document
+---
+
+```{r setup, echo = FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+library(tidyverse)
+library(knitr)
+library(ggplot2)
+library(dplyr)
+library(readr)
+```
+
+## Part 1: Tidying up the dad_mom file
+
+```{r show P1 data, echo = FALSE}
+dad_mom <- read.delim("dad_mom.txt", header = TRUE)
+kable(dad_mom, format = "markdown")
+### Just taking a look at what we're working with
+```
+
+```{r tidying, echo = FALSE}
+tidy_dad_mom <- dad_mom %>%
+  unite(dad, name_dad, income_dad) %>%
+  unite(mom, name_mom, income_mom) %>%
+### Put together all the mom data under 'mom' and dad data under 'dad', now to put it together into a chart
+  gather(key = "Parent", value = "value", mom, dad) %>%
+  separate("value", into = c("Name", "Income"))
+### Print out the results
+kable(tidy_dad_mom, format = "markdown")
+```
+
+## Part 2: Joining together CTRP data
+
+```{r show P2 data, echo = FALSE}
+AUC_1 <- read.csv("CTRP_files/AUC_1.csv")
+AUC_2 <- read.csv("CTRP_files/AUC_2.csv")
+ccell_info <- read.csv("CTRP_files/cancer_cell_line_info.csv")
+cmpd_info <- read.csv("CTRP_files/compound_info.csv")
+expt_info <- read.csv("CTRP_files/Experiment_info.csv")
+```
+
+```{r join, echo = FALSE}
+CTRP_total <- AUC_1 %>%
+  bind_rows(AUC_2) %>%
+  inner_join(expt_info, by = c("experiment_id" = "expt_id")) %>%
+  inner_join(ccell_info, by = "master_ccl_id") %>%
+  inner_join(cmpd_info, by = "master_cpd_id") 
+
+### print(CTRP_total)
+### Nevermind, don't print this out, the document is far too long
+```
+
+## Part 2 Q1: Which cancer type has the lowest AUC values to the compound "vorinostat"?
+
+```{r Q1: Which cancer type has the lowest AUC values to the compound "vorinostat"?, echo = TRUE}
+vorinostat_lowest <- CTRP_total %>%
+  filter(cpd_name == "vorinostat") %>%
+  arrange(area_under_curve)
+
+vorinostat_lowest %>%
+  ggplot() +
+  aes(area_under_curve, cancer_type) +
+  labs(title = "Vorinostat area under curve by cancer type", x = "Area under curve", y = "Cancer type") +
+  geom_boxplot()
+
+### print(vorinostat_lowest)
+### Seems like I can't get my boxplot to show up, not sure why. I manually printed out the data using print(vorinostat_lowest) and found that the lowest AUC values are predominantly "Haematopoietic and Lymphoid Tissues"
+```
```

Comment:
Just following up here, I realized I can run individual code chunks in rstudio. Running both my individual code chunk for Q1 and Q3 separately, I'm able to get the boxplots. Unfortunately, when I try 'knitting' the entire document, nothing shows up. Not sure what's going on here, will look at it some more when I have time!

Reply:
That is interesting that you were not able to get the boxplots after "knitting" the entire document. Could it be that there are some overlap of codes or something that is preventing the boxplot to appear after "knitting"? I checked both your codes for Q1 and Q3 and you had separate variables for it. I can't see why it would not run collectively, I am most definitely interested to know if you do find out why?

Reply:
This is really strange...did you try clearing your workspace / variables?  This hasn't happened to me in R yet but I know when working in MATLAB sometimes I have variables in my workspace that overlap with the code I am trying to run so usually clearing my workspace fixes it.